src/transformers/models/tapas/tokenization_tapas.py

```diff
@@ -2087,7 +2101,9 @@ def tokenize(self, text, never_split=None):
         # words in the English Wikipedia.).
         if self.tokenize_chinese_chars:
             text = self._tokenize_chinese_chars(text)
-        orig_tokens = whitespace_tokenize(text)
+        # prevents treating the same character with different unicode codepoints as different characters
+        unicode_normalized_text = unicodedata.normalize("NFC", text)
```

Comment:
Happy to add a new path, maybe `do_normalize` that's default false, or even ditch this part. This line was added to satisfy one more edge-case like part of test [here](https://github.com/huggingface/transformers/blob/abbc96a2148da0c91fa078bd021984b2cc10ef85/tests/models/clip/test_tokenization_clip.py#L92-L96), so we'd still get the thrust of the pr without it. Just lmk what's preferred.

I believe it at least wouldn't be an issue for CLIP since ftfy also does this by default on the text.

Reply:
I will let core maintainers take a stance on which direction we should go.

I think adding at least a test to show normalization happens is necessary. The link you provide https://github.com/huggingface/transformers/blob/abbc96a2148da0c91fa078bd021984b2cc10ef85/tests/models/clip/test_tokenization_clip.py#L92-L96 does not showcase that. It checks that both treat this utf-8 accentuation in the same way, not that normalization happens (which would required encoding the utf-8 accentuation and the normalized form and check they have the same id, and ideally explicitly mentionning the ID within the test).